plugins/google/compute/persistentDisksAutoDelete.js

```diff
@@ -0,0 +1,82 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'Persistent Disks Auto Delete',
+    category: 'Compute',
+    description: 'Ensure that auto-delete is disabled for attached persistent disks.',
+    more_info: 'When auto-delete is enabled, the attached persistent disk are deleted with VM instance deletion. In cloud environments, you might want to keep the attached persistent disks even when the associated VM instance is deleted.',
+    link: 'https://cloud.google.com/compute/docs/disks',
+    recommended_action: 'Ensure that auto-delete is disabled for all disks associated with your VM instances.',
+    apis: ['instances:compute:list', 'projects:get'],
+
+    run: function (cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions();
+
+        let projects = helpers.addSource(cache, source,
+            ['projects', 'get', 'global']);
+
+        if (!projects || projects.err || !projects.data || !projects.data.length) {
+            helpers.addResult(results, 3,
+                'Unable to query for projects: ' + helpers.addError(projects), 'global', null, null, (projects) ? projects.err : null);
+            return callback(null, results, source);
+        }
+
```

Comment:
We should user disks.list API and resource should be the actual disk instead of VM instance. I see that disks.list does not return autoDelete flag so we can first call VM instances API and get all disks which have autoDelete enabled as we are doing right now, and then call disks.list and check each disk against that list. Also we need to have a setting for disk_result_limit as in CSEK Encryption Enabled plugin.